<!-- -->
<launch>

    <arg name="project" default="emv_lio"/>
    <!-- Lidar odometry param -->
    <rosparam file="$(find emv_lio)/config/params_lidar.yaml" command="load" />
    
    <!-- VINS config -->
    <param name="vins_config_file" type="string" value="$(find emv_lio)/config/params_camera.yaml" />
    <param name="vins_config_file_1" type="string" value="$(find emv_lio)/config/params_camera1.yaml" />
    <param name="vins_config_file_2" type="string" value="$(find emv_lio)/config/params_camera2.yaml" />
    <param name="lio_config_file" type="string" value="$(find emv_lio)/config/params_lidar.yaml" />
    
    <!-- Lidar odometry -->
    <node pkg="$(arg project)"      type="$(arg project)_imuPreintegration"   name="$(arg project)_imuPreintegration"    output="screen"     respawn="true"/>
    <node pkg="$(arg project)"      type="$(arg project)_imageProjection"     name="$(arg project)_imageProjection"      output="screen"     respawn="true"/>
    <node pkg="$(arg project)"      type="$(arg project)_featureExtraction"   name="$(arg project)_featureExtraction"    output="screen"     respawn="true"/>
    <node pkg="$(arg project)"      type="$(arg project)_mapOptmization"      name="$(arg project)_mapOptmization"       output="screen"     respawn="true"/>

    <!-- Visual feature and odometry -->
    <node pkg="$(arg project)"      type="$(arg project)_visual_feature"      name="$(arg project)_visual_feature"       output="screen"     respawn="true"/>
    <node pkg="$(arg project)"      type="$(arg project)_visual_odometry"     name="$(arg project)_visual_odometry"      output="screen"     respawn="true"/>
    <node pkg="$(arg project)"      type="$(arg project)_visual_loop"         name="$(arg project)_visual_loop"          output="screen"     respawn="true"/>

    <!-- Image conversion -->
    <!-- <node pkg="image_transport" type="republish" name="$(arg project)_republish" args="compressed in:=/alphasense_driver_ros/cam0 raw out:=/alphasense_driver_ros/cam0" output="screen" respawn="true"/> -->

</launch>
